apis/v1alpha3/shared_types.go

```diff
@@ -0,0 +1,390 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import v1 "sigs.k8s.io/gateway-api/apis/v1"
+
+// ParentReference identifies an API object (usually a Gateway) that can be considered
+// a parent of this resource (usually a route). The only kind of parent resource
+// with "Core" support is Gateway. This API may be extended in the future to
+// support additional kinds of parent resources, such as HTTPRoute.
+//
+// Note that there are specific rules for ParentRefs which cross namespace
+// boundaries. Cross-namespace references are only valid if they are explicitly
+// allowed by something in the namespace they are referring to. For example:
+// Gateway has the AllowedRoutes field, and ReferenceGrant provides a
+// generic way to enable any other kind of cross-namespace reference.
+//
+// The API object must be valid in the cluster; the Group and Kind must
+// be registered in the cluster for this reference to be valid.
+// +k8s:deepcopy-gen=false
+type ParentReference = v1.ParentReference
```

Comment:
On further thought, since 100% of the types in this file are already in v1, it's probably best to remove this file and replace it with references to shared v1 types wherever applicable.